src/models/connectionCredentials.ts

```diff
@@ -214,8 +218,13 @@ export class ConnectionCredentials implements IConnectionCredentials {
         return utils.isEmpty(credentials.user) && ConnectionCredentials.isPasswordBasedCredential(credentials);
     }
 
+    // Prompt for password if this is a password based credential and the password for the profile was empty
+    // and not explicitly set as empty. If it was explicitly set as empty, only prompt if pw not saved
     private static shouldPromptForPassword(credentials: IConnectionCredentials): boolean {
-        return utils.isEmpty(credentials.password) && ConnectionCredentials.isPasswordBasedCredential(credentials);
+        return utils.isEmpty(credentials.password)
+            && ConnectionCredentials.isPasswordBasedCredential(credentials)
+            && (!((<IConnectionProfile>credentials).emptyPasswordInput) ||
+             ((<IConnectionProfile>credentials).emptyPasswordInput && !(<IConnectionProfile>credentials).savePassword));
```

Comment:
`<IConnectionProfile>credentials).emptyPasswordInput` is redundant here since the first part of the or clause checks it